_blog/2018/v1alpha3-routing.md

```diff
@@ -0,0 +1,365 @@
+   `DestinationRule`, described in the following section.
+3. `VirtualService` hosts can be specified using wildcard DNS prefixes to create a single rule for all matching services.
+   For example, in Kubernetes, to apply the same rewrite rule for all services in the `foo` namespace, the `VirtualService`
+   would use `*.foo.svc.cluster.local` as the host.
+
+### DestinationRule
+
+A `DestinationRule` configures the set of policies to be applied at a destination after routing has occurred. They are
+intended to be authored by service owners, describing the circuit breakers, load balancer settings, TLS settings, etc.. 
+`DestinationRule` is more or less the same as its predecessor, `DestinationPolicy`, with the following exceptions:
+
+1. The `host` of a `DestinationRule` can include wildcard prefixes, allowing a single rule to be specified for many actual
+   services.
+2. A `DestinationRule` defines addressable `subsets` (i.e., named versions) of the corresponding destination host. These
+   subsets are used in `VirtualService` route specifications when sending traffic to specific versions of the service.
+   Naming versions this way allows us to cleanly refer to them across different virtual services, simplify the stats that
+   Istio proxies emit, and to encode subsets in SNI headers.
+
+A `DestinationRule` that configures policies and subsets for the reviews service might look something like this:
+
+```yaml
+apiVersion: networking.istio.io/v1alpha3
+kind: DestinationRule
+metadata:
+  name: reviews
+spec:
+  host: reviews
+  trafficPolicy:
+    loadBalancer:
+      simple: RANDOM
+  subsets:
+  - name: v1
+    labels:
+      version: v1
+  - name: v2
+    labels:
+      version: v2
+    trafficPolicy:
+      loadBalancer:
+        simple: ROUND_ROBIN
+  - name: v3
+    labels:
+      version: v3
+```
+
+Notice that, unlike `DestinationPolicy`, multiple policies (e.g., default and v2-specific) are specified in a single
+`DestinationRule` configuration.
+
+### ExternalService
+
+`ExternalService` is used to add additional entries into the service registry that Istio maintains internally. This allows
+one to model traffic to external dependencies of the mesh such as APIs consumed from the web or traffic to services in
+legacy infrastructure.
+
+Replacing `EgressRule` from the previous API, `ExternalService` provides several significant improvements, such as:
+
+1. Individual service ports and service discovery mode are now configurable.
+2. One or more (different) external endpoints can be configured to implement a “virtual” external service.
+3. Secure HTTP services (automatic TLS upgrade) can now be accessed using standard https (e.g., `https://secureservice.com/`
+   instead of `http://secureservice.com:443/`.
+4. Multiple CIDR subsets can now be included in a single `ExternalService` configuration.
+
+Because an `ExternalService` configuration simply adds an external destination to the internal service registry, it can be
```

Comment:
Maybe we should mention ExternalService is the new way to represent 'mesh expansion', i.e. VMs running istio outside a cluster but still as part of the mesh.

Reply:
Are you sure you want to go down this path? The code to treat external services is different. It does not set up istio auth. You have to configure tls using destination rules. Mesh expansion assumes that services are part of the mesh and honor auth, mixer, etc. 

External service assumes that traffic is going to potentially untrusted/not in istio control, entities. The mixer configuration is different as well.

I would be skeptical of overriding this behavior with dual semantics. Put another way, being part of service registry is not equivalent to being part of the mesh. 

Why not add a simple MeshService resource that allows you to statically add services to the mesh ? Or add label annotations to endpoints that allow you to supply label info for vm endpoints ( that’s all you seem to want out of external services).

Cc @louiscryan wdyt of adding label annotations to vm endpoints ? If you think you need a cross platform solution, then let’s add an explicit resource that models the intent of an in-mesh entity